src/lazy/text/matched.rs

```diff
@@ -383,19 +389,25 @@ impl MatchedString {
     // Strings longer than 64 bytes will allocate a larger space on the heap.
     const STACK_ALLOC_BUFFER_CAPACITY: usize = 64;
 
-    pub fn read<'data>(&self, matched_input: TextBufferView<'data>) -> IonResult<StrRef<'data>> {
+    pub fn read<'data>(
+        &self,
+        allocator: &'data BumpAllocator,
```

Comment:
🗺️ Text escapes and base64 decoding now happen in bump-allocated scratch space instead of heap-allocated `String`s and `Vec`s.